src/Tempest/Http/src/RouteConfig.php

```diff
@@ -29,7 +34,14 @@ public function addRoute(MethodReflector $handler, Route $route): self
         if ($route->isDynamic) {
             $this->regexMark = str_increment($this->regexMark);
             $this->dynamicRoutes[$route->method->value][$this->regexMark] = $route;
```

Comment:
Do we still need these two separate arrays if we have the routing tree? Don't worry about refactoring it in this PR if the answer is "no", but I did wanna check up on it

Reply:
Yes. The routing tree only facilitates building up the routing regex for the dynamic routes.